core/src/main/scala/kafka/log/Log.scala

```diff
@@ -106,6 +106,7 @@ case class LogAppendInfo(var firstOffset: Option[Long],
                          var logAppendTime: Long,
                          var logStartOffset: Long,
                          var recordConversionStats: RecordConversionStats,
+                         var rolled: Boolean,
```

Comment:
I implemented this suggestion instead of using `rolled`. We discuss this offline and is it documented in the code. The caveat in the implementation is that the segment's base offset and relative offset in the segment is unknown when the appended records are duplicates.